src/transactions/TransactionFrame.cpp

```diff
@@ -1780,7 +1791,9 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
                 {
                     changes = ltxOp.getChanges();
                 }
-                operationMetas.emplace_back(changes);
+                xdr::xvector<ContractEvent> ces;
```

Comment:
I'm not a fan of all these moves and flushes, as they leave the data structures in invalid state (and unlike in Rust it's easy to mis-use them accidentally). I don't think we need to block the PR on this though (besides the small issues that I've mentioned in the other comments), but I'll probably try to make another pass on it after its merge in order to reduce the number of necessary copies.